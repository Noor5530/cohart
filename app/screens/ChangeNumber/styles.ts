import { appFonts } from 'components/text';
import { iconColor, textColor } from 'config/colors';
import { StyleSheet } from 'react-native';
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from 'react-native-responsive-screen';

const style = StyleSheet.create({
  container: {
    height: hp(100),
    width: '100%',
    backgroundColor: textColor,
  },
  header: {
    flex: 1,
    width: '100%',
  },
  mainContainer: {
    flex: 1,
    width: '100%',
    paddingHorizontal: wp('5'),
  },
  bottomContainer: { flex: 1, width: '100%', alignItems: 'center' },
  button: {
    marginTop: hp(5),
    height: 35,
    width: 140,
    alignSelf: 'center',
    borderRadius: 30,
    borderWidth: 1,
    borderColor: iconColor,
    alignItems: 'center',
    justifyContent: 'center',
  },
  bottonText: {
    fontWeight: '500',
    fontFamily: appFonts.InterRegular,
    fontSize: 13,
  },
  bottomText: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    height: 40,
  },
  cross: {
    width: 20,
    height: 20,
    borderRadius: 10,
    borderWidth: 1,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 10,
  },
  label: {
    fontSize: 13,
    color: iconColor,
  },
  reefersText: {
    paddingTop: 25,
    height: 55,
    textAlignVertical: 'top',
    fontSize: 13,
    color: iconColor,
  },
  textInput: {
    height: 50.42,
    padding: 0,
    margin: 0,
    width: '100%',
    color: iconColor,
    fontFamily: appFonts.InterRegular,
    fontSize: wp('5'),
    fontWeight: 'bold',
    backgroundColor: textColor,
    borderWidth: 0,
  },
  textContainer: {
    width: '100%',
    margin: 0,
    backgroundColor: textColor,
    height: 50.42,
  },
  countryContainerStyle: {
    borderWidth: 2,
    borderColor: iconColor,
    borderRadius: 10,
    paddingHorizontal: 10,
    backgroundColor: textColor,
  },
  codeTextStyle: {
    margin: 0,
    color: iconColor,
    fontFamily: appFonts.InterRegular,
    fontSize: wp('5'),
    fontWeight: 'bold',
  },
  inputContainerStyle: {
    padding: 0,
    margin: 0,
    width: '100%',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    height: 50.42,
    borderColor: '#000000',
    borderRadius: 14.4067,
    backgroundColor: textColor,
  },
  heading: {
    paddingVertical: 20,
    fontFamily: appFonts.InterRegular,
    color: iconColor,
    fontWeight: '400',
    fontSize: 15,
  },
  changeNumberContainer: {
    flex: 1,
    width: '100%',
    justifyContent: 'center',
    paddingTop: hp(6),
  },
  numberInput: { flexDirection: 'row', height: 51 },
  changeNumberButton: { flex: 1 },
});
export default style;
