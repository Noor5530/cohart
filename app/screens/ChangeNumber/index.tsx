import { useIsFocused, useNavigation } from '@react-navigation/native';
import analytics from '@segment/analytics-react-native';
import ChangePhoneNumber from 'assets/ChangePhoneNumber.png';
import StatusBar from 'components/CustomStatusBar';
import Header from 'components/Header';
import HeaderHeading from 'components/HeaderHeading';
import PhoneTextInput from 'components/PhoneModel';
import ApiConfig from 'config/api-config';
import { iconColor, profileColor, textColor } from 'config/colors';
import useFetch from 'hooks/useFetch';
import { isIphone } from 'lib/isIphone';
import AppState from 'models/reducers';
import React, { useEffect, useRef, useState } from 'react';
import { Alert, Text, TextInput, TouchableOpacity, View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { useDispatch, useSelector } from 'react-redux';
import { disableLoading, enableLoading } from 'store/actions/loadingActions';

import { appFonts } from '../../components/text';
import { enableSnackBar } from '../../store/actions/snackBarAction';
import { changeNumberRequest } from '../../store/actions/userActions';
import styles from './styles';

const ChangeNumber = () => {
    const phone_number = useSelector(
        (state: AppState) => state.user.phone_number,
    );
    const user = useSelector((state: AppState) => state.user);

    const [code, setCode] = useState('');
    const userId = useSelector((state: AppState) => state.user._id);
    const [number, setNumber] = useState('');
    const dispatch = useDispatch();
    const [updatedPhoneNumber, setUpdatedPhoneNumber] = React.useState('');
    const [error, setError] = useState('');
    const [codeError, setSetCodeError] = useState('');

    const [isUpdateNumber, setIsUpdateNumber] = useState(false);
    const navigation = useNavigation();
    const isFocused = useIsFocused();
    const {
      state: { status, data: sendCodeResponse },
      refetch: sendCode,
    } = useFetch<any>(ApiConfig.SEND_SMS_FOR_NUMBER_CHANGE, {
      method: "POST",
      data: {
        old_phone_number: phone_number,
        new_phone_number: updatedPhoneNumber,
      },
    });

    useEffect(() => {
        if (!isFocused) {
            setNumber('');
            setUpdatedPhoneNumber('');
            setError('');
            setIsUpdateNumber(false);
            setCode('');
            setSetCodeError('');
        }
    }, [isFocused]);

    const onSubmit = () => {
        const regex = /^\+(?:[0-9] ?){6,14}[0-9]$/;

        if (regex.test(updatedPhoneNumber)) {
            setError('');
            Alert.alert('Change Number', 'Are you sure ?', [
                {
                    text: 'Cancel',
                    onPress: () => console.log('Cancel Pressed'),
                    style: 'cancel',
                },
                {
                    text: 'OK',
                    onPress: () => {
                        //      setIsUpdateNumber(true);
                        sendCode({
                            data: {
                                old_phone_number: phone_number,
                                new_phone_number: updatedPhoneNumber,
                            },
                        });
                    },
                },
            ]);
        } else {
            setError('Please enter valid Number');
        }
    };

    useEffect(() => {
        if (status == 'fetching') {
            dispatch(enableLoading());
        } else if (status == 'fetched' && sendCodeResponse.statusCode === 200) {
            setIsUpdateNumber(true);
            dispatch(disableLoading());
        } else if (status !== 'init') {
            dispatch(disableLoading());
            setError(sendCodeResponse?.data ? sendCodeResponse?.data : '');

            dispatch(enableSnackBar(sendCodeResponse?.data));
        }
    }, [status, sendCodeResponse]);
    const onSavedNumber = () => {
        if (code == sendCodeResponse.data.sms_code) {
            analytics.track('Changed_phone_number', {
                new_phone_number: updatedPhoneNumber,
                current_phone_number: phone_number,
                user_id: userId,
                sms_code: code,
            });
            dispatch(
                changeNumberRequest({
                    new_phone_number: updatedPhoneNumber,
                    current_phone_number: phone_number,
                    user_id: userId,
                    sms_code: code,
                    country_code: updatedPhoneNumber.slice(0, -number.length),
                    number: number,
                }),
            );
        } else {
            setSetCodeError('Code is not valid');
        }
    };
    const textInputRef = useRef();
    return (
        <View style={styles.container} key="changeNumber">
            <StatusBar backgroundColor={textColor} barStyle="dark-content" />
            <Header
                style={{
                    width: '100%',
                    paddingTop: isIphone(),
                    borderBottomColor: iconColor,
                    borderBottomWidth: 1.5,
                }}
                onPressStar={() => {
                    if (isUpdateNumber) {
                        setIsUpdateNumber(false);
                    } else {
                        navigation.goBack();
                    }
                }}
                textStyle={{
                    color: iconColor,
                    fontFamily: appFonts.AktivGroteskEx_Trial_Bd,
                }}
                back
                color={iconColor}
            />
            <HeaderHeading
                drawer={false}
                title={'Change\nPhone #'}
                icon={ChangePhoneNumber}
            />
            <KeyboardAwareScrollView
                style={styles.mainContainer}
                contentContainerStyle={{ paddingBottom: hp('10') }}>
                <View style={styles.changeNumberContainer}>
                    {!isUpdateNumber && (
                        <>
                            <Text style={styles.heading}>Current Number</Text>
                            <View style={styles.numberInput}>
                                <View
                                    style={[
                                        styles.countryContainerStyle,
                                        { justifyContent: 'center' },
                                    ]}>
                                    <Text style={styles.codeTextStyle}>
                                        {user?.country_code
                                            ? user?.country_code
                                            : ''}
                                    </Text>
                                </View>
                                <TextInput
                                    maxLength={14}
                                    editable={false}
                                    value={
                                        user?.user_number
                                            ? user?.user_number
                                            : ''
                                    }
                                    style={[
                                        styles.textInput,
                                        { paddingLeft: 15 },
                                    ]}
                                />
                            </View>
                        </>
                    )}
                    {isUpdateNumber && (
                        <Text style={[styles.heading, { textAlign: 'center' }]}>
                            Please enter verification code we sent to your
                            mobile
                        </Text>
                    )}
                    <Text style={styles.heading}>New Number</Text>

                    <PhoneTextInput
                        ref={textInputRef}
                        defaultCode={'US'}
                        defaultValue={number}
                        onChangeFormattedText={text => {
                            setUpdatedPhoneNumber(text);
                        }}
                        maxLength={20}
                        value={number}
                        textInputStyle={styles.textInput}
                        textContainerStyle={styles.textContainer}
                        countryContainerStyle={styles.countryContainerStyle}
                        codeTextStyle={styles.codeTextStyle}
                        containerStyle={styles.inputContainerStyle}
                        placeholder="xxxxxxxxxx"
                        disableArrowIcon={true}
                        layout="second"
                        onChangeText={text => {
                            setNumber(text);
                            setError('');
                        }}
                    />
                    {error !== '' && (
                        <Text style={{ color: iconColor, fontSize: 12 }}>
                            {error}
                        </Text>
                    )}
                </View>
                {isUpdateNumber && (
                    <>
                        <TextInput
                            value={code}
                            onChangeText={text => {
                                setSetCodeError('');
                                setCode(text);
                            }}
                            maxLength={6}
                            placeholder={'XXXXXX'}
                            style={[
                                styles.textInput,
                                {
                                    marginTop: hp(5),

                                    borderWidth: 2,
                                    borderColor: iconColor,
                                    width: '50%',
                                    alignSelf: 'center',
                                    alignItems: 'center',
                                    textAlign: 'center',
                                    borderRadius: 14.4067,
                                },
                            ]}
                        />
                        {codeError !== '' && (
                            <Text
                                style={{
                                    paddingTop: 5,
                                    textAlign: 'center',
                                    color: iconColor,
                                    fontSize: 12,
                                }}>
                                {codeError}
                            </Text>
                        )}
                        <Text
                            style={[
                                styles.heading,
                                {
                                    textAlign: 'center',
                                    paddingBottom: 0,
                                    paddingTop: 10,
                                },
                            ]}>
                            Didn’t receive it?{' '}
                        </Text>
                        <TouchableOpacity
                            onPress={() => {
                                sendCode({
                                    data: {
                                        old_phone_number: phone_number,
                                        new_phone_number: updatedPhoneNumber,
                                    },
                                });
                            }}>
                            <Text
                                onPress={() => {
                                    sendCode({
                                        data: {
                                            old_phone_number: phone_number,
                                            new_phone_number:
                                                updatedPhoneNumber,
                                        },
                                    });
                                }}
                                style={[
                                    styles.heading,
                                    {
                                        paddingVertical: 5,
                                        textAlign: 'center',
                                        color: profileColor,
                                    },
                                ]}>
                                Resend Code
                            </Text>
                        </TouchableOpacity>
                    </>
                )}
                {isUpdateNumber ? (
                    <TouchableOpacity
                        onPress={onSavedNumber}
                        style={styles.button}>
                        <Text style={styles.bottonText}>Verify</Text>
                    </TouchableOpacity>
                ) : (
                    <TouchableOpacity onPress={onSubmit} style={styles.button}>
                        <Text style={styles.bottonText}>CHANGE</Text>
                    </TouchableOpacity>
                )}
            </KeyboardAwareScrollView>
        </View>
    );
};
export default React.memo(ChangeNumber);
