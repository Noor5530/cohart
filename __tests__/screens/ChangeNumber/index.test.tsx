import 'react-native';
import React from 'react'; 
import ChangeNumber from '../../../app/screens/ChangeNumber';
import renderer from 'react-test-renderer';

jest.useFakeTimers()

test('renders correctly', () => {

    const tree = renderer.create(
        <ChangeNumber />
    ).toJSON();

    expect(tree).toMatchSnapshot();

});
