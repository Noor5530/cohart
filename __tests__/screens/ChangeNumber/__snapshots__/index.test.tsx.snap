// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#000000",
        "borderBottomWidth": 1.5,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingTop": 30,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "width": 38,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="14"
          bbWidth="21"
          fill="none"
          focusable={false}
          height="14"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 14,
                "width": 21,
              },
            ]
          }
          vbHeight={14}
          vbWidth={21}
          width="21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M6.64474 0.583984L0.433134 6.79559L0.167969 7.06076L0.433134 7.32592L6.64474 13.5375L7.17507 13.0072L1.66787 7.5L21 7.5V6.75L1.53939 6.75L7.17507 1.11431L6.64474 0.583984Z"
              fill={4278190080}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 20,
            },
            Object {
              "color": "#000000",
              "fontFamily": "AktivGroteskExTrial-Bold",
            },
            Object {
              "fontFamily": "AktivGroteskExTrial-Bold",
            },
          ]
        }
      >
        COHART
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="19"
        bbWidth="16"
        fill="none"
        focusable={false}
        height="19"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 19,
              "width": 16,
            },
          ]
        }
        vbHeight={19}
        vbWidth={16}
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M7.58726 1.75C8.01448 1.75 8.36081 2.09633 8.36081 2.52355C8.36081 3.03874 8.66732 3.59741 9.26471 3.7987C11.462 4.53906 13.0903 6.71461 13.0903 9.32316C13.0903 9.36104 13.0899 9.39882 13.0893 9.43649C13.0881 9.50183 13.0931 9.56582 13.1039 9.62789V12.5105C13.1039 12.8394 13.1805 13.1637 13.3275 13.4579L14.1596 15.1219C14.199 15.2007 14.1417 15.2934 14.0536 15.2934H1.11992C1.03185 15.2934 0.974564 15.2007 1.01395 15.1219L1.81191 13.526C1.95899 13.2319 2.03556 12.9075 2.03556 12.5786V9.75294C2.06936 9.64853 2.08682 9.53692 2.08501 9.42124C2.0845 9.38862 2.08424 9.35593 2.08424 9.32316C2.08424 6.71462 3.71252 4.53907 5.90981 3.7987C6.5072 3.59741 6.81371 3.03874 6.81371 2.52355C6.81371 2.09633 7.16004 1.75 7.58726 1.75ZM9.58401 2.85105C9.44664 2.80476 9.36081 2.66851 9.36081 2.52355C9.36081 1.54405 8.56676 0.75 7.58726 0.75C6.60775 0.75 5.81371 1.54405 5.81371 2.52355C5.81371 2.66851 5.72787 2.80476 5.5905 2.85105C2.97586 3.73204 1.08424 6.29665 1.08424 9.32316C1.08424 9.36116 1.08454 9.39908 1.08513 9.43693C1.08535 9.45079 1.07419 9.46217 1.06033 9.46217C1.04665 9.46217 1.03556 9.47326 1.03556 9.48694V12.5786C1.03556 12.7523 0.995133 12.9235 0.91748 13.0788L0.119523 14.6747C-0.252313 15.4184 0.288463 16.2934 1.11992 16.2934H5.15932C5.40744 17.4128 6.406 18.25 7.6001 18.25C8.79419 18.25 9.79276 17.4128 10.0409 16.2934H14.0536C14.8851 16.2934 15.4258 15.4184 15.054 14.6747L14.222 13.0107C14.1443 12.8554 14.1039 12.6841 14.1039 12.5105V9.46957C14.1039 9.46549 14.1006 9.46217 14.0965 9.46217C14.0923 9.46217 14.089 9.45878 14.0891 9.45464C14.0899 9.41091 14.0903 9.36709 14.0903 9.32316C14.0903 6.29665 12.1987 3.73203 9.58401 2.85105ZM8.99864 16.2934H6.20155C6.41923 16.8532 6.96332 17.25 7.6001 17.25C8.23687 17.25 8.78097 16.8532 8.99864 16.2934Z"
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderBottomWidth": 2,
        "borderColor": "#000000",
        "flexDirection": "row",
        "paddingTop": 35,
        "width": "90%",
      }
    }
  >
    <Text
      numberOfLines={3}
      style={
        Object {
          "fontFamily": "Inter-Bold",
          "fontSize": undefined,
          "maxWidth": "90%",
          "textAlign": "left",
        }
      }
    >
      Change
Phone #
    </Text>
    <View
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/ChangePhoneNumber.png",
          }
        }
        style={
          Object {
            "height": undefined,
            "marginTop": 5,
            "width": undefined,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingTop": undefined,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Inter-Regular",
          "fontSize": 15,
          "fontWeight": "400",
          "paddingVertical": 20,
        }
      }
    >
      Current Number
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 51,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#000000",
              "borderRadius": 10,
              "borderWidth": 2,
              "paddingHorizontal": 10,
            },
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Inter-Regular",
              "fontSize": undefined,
              "fontWeight": "bold",
              "margin": 0,
            }
          }
        >
          
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={false}
        maxLength={14}
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderWidth": 0,
              "color": "#000000",
              "fontFamily": "Inter-Regular",
              "fontSize": undefined,
              "fontWeight": "bold",
              "height": 50.42,
              "margin": 0,
              "padding": 0,
              "width": "100%",
            },
            Object {
              "paddingLeft": 15,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Inter-Regular",
          "fontSize": 15,
          "fontWeight": "400",
          "paddingVertical": 20,
        }
      }
    >
      New Number
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "flexDirection": "row",
            "width": 600,
          },
          Object {},
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#000000",
            "borderRadius": 14.4067,
            "height": 50.42,
            "justifyContent": "flex-start",
            "margin": 0,
            "padding": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#000000",
            "borderRadius": 10,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 10,
          }
        }
      >
        <mockedDefaultExport
          countryCode="US"
          filterProps={Object {}}
          onClose={[Function]}
          onSelect={[Function]}
          renderFlagButton={[Function]}
          visible={false}
          withCallingCode={true}
          withEmoji={true}
          withFilter={true}
          withFlag={true}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": undefined,
                "fontWeight": "bold",
                "margin": 0,
              },
            ]
          }
        >
          +undefined
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F8F9F9",
              "flex": 1,
              "flexDirection": "row",
              "paddingHorizontal": 30,
              "paddingVertical": 27,
              "textAlign": "left",
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "height": 50.42,
              "margin": 0,
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardAppearance="default"
          keyboardType="number-pad"
          maxLength={20}
          onChangeText={[Function]}
          placeholder="xxxxxxxxxx"
          rejectResponderTermination={true}
          selectionColor="black"
          style={
            Array [
              Object {
                "color": "#000000",
                "flex": 1,
                "fontSize": 16,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderWidth": 0,
                "color": "#000000",
                "fontFamily": "Inter-Regular",
                "fontSize": undefined,
                "fontWeight": "bold",
                "height": 50.42,
                "margin": 0,
                "padding": 0,
                "width": "100%",
              },
            ]
          }
          testID="phoneInputText"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#000000",
        "borderRadius": 30,
        "borderWidth": 1,
        "height": 35,
        "justifyContent": "center",
        "marginTop": undefined,
        "opacity": 1,
        "width": 140,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Inter-Regular",
          "fontSize": 13,
          "fontWeight": "500",
        }
      }
    >
      CHANGE
    </Text>
  </View>
</View>
`;
